public/app.d.ts

```diff
@@ -1706,42 +1706,505 @@ declare namespace JMap {
+       * JMap.Application.Print.deactivateQuery()
+       * ```
+       */
       function deactivateQuery(): void
-      function setDefaultData(values: any): void
+      
+      /**
+       * **JMap.Application.Query.getDefaultData**
+       *
+       * Returns the current query form data.
+       * 
+       * If there is no active form, warn in the console and return an empty object {}.
+       * 
+       * @example ```ts
+       * 
+       * // returns the current query form data
+       * JMap.Application.Print.getDefaultData()
+       * ```
+       */
       function getDefaultData(): { [ id: string ]: any }
+
+      /**
+       * **JMap.Application.Query.setDefaultData**
+       *
+       * Set active query form data. If there is no active form, this function throws.
+       * 
+       * @throws if no query is active
+       * @param values the form data, depends on the form
+       * @example ```ts
+       * 
+       * // set the current query form data
+       * JMap.Application.Print.setDefaultData({
+       *  $param1: "black",
+       *  $param2: "white"
+       * })
+       * ```
+       */
+      function setDefaultData(values: any): void
+
+      /**
+       * **JMap.Application.Query.clearDefaultData**
+       *
+       * Reset the current query form data.
+       * 
+       * If there is no active form, do nothing.
+       * 
+       * @example ```ts
+       * 
+       * // clear the current query form data
+       * JMap.Application.Print.clearDefaultData()
+       * ```
+       */
       function clearDefaultData(): void
+
+      /**
+       * **JMap.Application.Query.displayInDialog**
+       *
+       * By default the query form is displayed in the left panel with a simple layout, one field per line.
+       * 
+       * But it is possible to display it in a modal window, with the layout defined in the administration.
+       * 
+       * @throws if isVisibleInDialog is not a boolean
+       * @param isVisibleInDialog true in modal window, false in left panel
+       * @example ```ts
+       * 
+       * // display the current form in a modal window
+       * JMap.Application.Print.displayInDialog(true)
+       * 
+       * // display the current form in the left panel
+       * JMap.Application.Print.displayInDialog(false)
+       * ```
+       */
       function displayInDialog(isVisibleInDialog: boolean): void
-      function processQuery(values: any): Promise<void>
+
+      /**
+       * **JMap.Application.Query.processQuery**
+       *
+       * Process active query form with the given values.  
+       * 
+       * Throws if no query is active.
+       * 
+       * Resolve only if some features are found.
+       * 
+       * If features are found, we make the layer selectable and visible if needed, then select and zoom on found features.
+       * 
+       * If no feature are found, reject promise. When rejected, it rejects human readable error, using user locale.
```

Comment:
If no feature is found, the Promise rejects with a human readable error in the current user's locale.